warehouse/jobs/runner.go

```diff
@@ -35,75 +36,52 @@ func InitWarehouseJobsAPI(
 }
 
 func WithConfig(a *AsyncJobWh, config *config.Config) {
-	a.MaxBatchSizeToProcess = config.GetInt("Warehouse.jobs.maxBatchSizeToProcess", 10)
-	a.MaxCleanUpRetries = config.GetInt("Warehouse.jobs.maxCleanUpRetries", 5)
-	a.MaxQueryRetries = config.GetInt("Warehouse.jobs.maxQueryRetries", 3)
-	a.MaxAttemptsPerJob = config.GetInt("Warehouse.jobs.maxAttemptsPerJob", 3)
-	a.RetryTimeInterval = config.GetDuration("Warehouse.jobs.retryTimeInterval", 10, time.Second)
-	a.AsyncJobTimeOut = config.GetDuration("Warehouse.jobs.asyncJobTimeOut", 300, time.Second)
+	a.maxBatchSizeToProcess = config.GetInt("Warehouse.jobs.maxBatchSizeToProcess", 10)
+	a.maxCleanUpRetries = config.GetInt("Warehouse.jobs.maxCleanUpRetries", 5)
+	a.maxQueryRetries = config.GetInt("Warehouse.jobs.maxQueryRetries", 3)
+	a.maxAttemptsPerJob = config.GetInt("Warehouse.jobs.maxAttemptsPerJob", 3)
+	a.retryTimeInterval = config.GetDuration("Warehouse.jobs.retryTimeInterval", 10, time.Second)
+	a.asyncJobTimeOut = config.GetDuration("Warehouse.jobs.asyncJobTimeOut", 300, time.Second)
 }
 
 func (a *AsyncJobWh) getTableNamesBy(sourceID, destinationID, jobRunID, taskRunID string) ([]string, error) {
 	a.logger.Infof("[WH-Jobs]: Extracting table names for the job run id %s", jobRunID)
 	var tableNames []string
 	var err error
-	query := `SELECT id
-		FROM
-		` + warehouseutils.WarehouseUploadsTable + `
-		WHERE metadata->>'source_job_run_id'=$1
-			AND metadata->>'source_task_run_id'=$2
-			AND metadata in (
-				SELECT
-					metadata
-				FROM
-				` + warehouseutils.WarehouseUploadsTable + `
-				WHERE source_id=$3
-				AND destination_id=$4
-			)`
+
+	query := `SELECT table_name FROM ` + warehouseutils.WarehouseTableUploadsTable + ` WHERE wh_upload_id IN ` +
+		` (SELECT id FROM ` + warehouseutils.WarehouseUploadsTable + ` WHERE metadata->>'source_job_run_id'=$1
+				AND metadata->>'source_task_run_id'=$2
+				AND source_id=$3
+				AND destination_id=$4)`
```

Comment:
Do we have indexes at least on `source_id` and `destination_id`? I remember seeing some performance improvements in other DBs by querying by those columns first (i.e. starting the `WHERE` clause with the indexed columns). Perhaps the Postgres query planner is smart enough that it does it on its own but probably worth double checking:

```
SELECT x, y FROM t1 WHERE indexed_col1 = 1 AND indexed_col = 2 AND json_stuff_here_without_indexes
```

Reply:
We do have an index `wh_uploads_source_destination_id_index`